<?xml version="1.0"?>
<launch>
	<!-- MAVROS posix SITL environment launch script -->
	<!-- launches MAVROS, PX4 SITL, Gazebo environment, and spawns vehicle -->
	<!-- vehicle pose -->
	<arg name="x" default="0"/>
	<arg name="y" default="0"/>
	<arg name="z" default="0"/>
	<arg name="R" default="0"/>
	<arg name="P" default="0"/>
	<arg name="Y" default="0"/>
	<!-- <arg name="Y" default="1.5"/> -->
	<!-- vehicle model and world -->
	<arg name="est" default="ekf2"/>
	<arg name="vehicle" default="rpl_uav_depth_camera"/>
	<arg name="world" default="$(find simulation)/worlds/aep_office.world"/>
	<arg name="sdf" default="$(find simulation)/models/$(arg vehicle)/$(arg vehicle).sdf"/>
	<!-- <arg name="sdf" default="$(find mavlink_sitl_gazebo)/models/iris/iris.sdf"/> -->
	<arg name="config_yaml" default="$(find simulation)/config/px4_config.yaml" />

	<!-- gazebo configs -->
	<arg name="gui" default="false"/>
	<arg name="debug" default="false"/>
	<arg name="verbose" default="true"/>
	<arg name="paused" default="false"/>
	<arg name="respawn_gazebo" default="false"/>
	<!-- MAVROS configs -->
	<arg name="fcu_url" default="udp://:14540@localhost:14557"/>
	<arg name="gcs_url" default="" />
	<arg name="tgt_system" default="1" />
	<arg name="tgt_component" default="1" />
	<arg name="log_output" default="screen" />
	<arg name="fcu_protocol" default="v2.0" />
	<arg name="respawn_mavros" default="false"/>
	<!-- PX4 configs -->
	<arg name="interactive" default="false"/>
	<!-- PX4 SITL and Gazebo -->
	<include file="$(find px4)/launch/posix_sitl.launch">
		<arg name="x" value="$(arg x)"/>
		<arg name="y" value="$(arg y)"/>
		<arg name="z" value="$(arg z)"/>
		<arg name="R" value="$(arg R)"/>
		<arg name="P" value="$(arg P)"/>
		<arg name="Y" value="$(arg Y)"/>
		<arg name="world" value="$(arg world)"/>
		<arg name="vehicle" value="iris"/>
		<arg name="sdf" value="$(arg sdf)"/>
		<arg name="gui" value="$(arg gui)"/>
		<arg name="interactive" value="$(arg interactive)"/>
		<arg name="debug" value="$(arg debug)"/>
		<arg name="verbose" value="$(arg verbose)"/>
		<arg name="paused" value="$(arg paused)"/>
		<arg name="respawn_gazebo" value="$(arg respawn_gazebo)"/>
	</include>
	<!-- MAVROS -->
	<include file="$(find mavros)/launch/node.launch">
		<arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
		<arg name="config_yaml" value="$(arg config_yaml)" />

		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" value="$(arg gcs_url)"/>
		<arg name="tgt_system" value="$(arg tgt_system)" />
		<arg name="tgt_component" value="$(arg tgt_component)" />
		<arg name="log_output" value="$(arg log_output)" />
		<arg name="fcu_protocol" value="$(arg fcu_protocol)" />
		<arg name="respawn_mavros" default="$(arg respawn_mavros)" />
	</include>

	<!-- <node name="ground_truth_tf_broadcaster" pkg="simulation" type="ground_truth_tf_broadcaster_node" /> -->

	<node name="stabilized_transform_publisher" pkg="simulation" type="stabilized_transform_publisher_node" output="screen" required="false">
		<remap from="odom" to="/base_link_ground_truth" />
		<param name="map_frame_id" value="map"/>
		<param name="stabilized_frame_id" value="base_stabilized_link" />
		<param name="robot_frame_id" value="base_link" />
	</node>

	<param name="robot_description" textfile="$(find simulation)/urdf/$(arg vehicle).urdf" />
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen" required="false" />
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" output="screen" required="false">
		<!-- TODO -->
	</node>

	<rosparam command="load" file="$(find simulation)/config/iris.yaml" ns="/mavros/param" />
</launch>
